x-pack/legacy/plugins/apm/server/lib/transactions/breakdown/index.ts

```diff
@@ -93,7 +95,7 @@ export async function getTransactionBreakdown({
             {
               term: {
                 [TRANSACTION_TYPE]: {
-                  value: 'request'
+                  value: transactionType
```

Comment:
nit: I like to use the shorthand of the terms agg, and avoid ternaries when there are conditionals:

```ts
const filters = [
  { term: { [SERVICE_NAME]: serviceName } },
  { term: { [TRANSACTION_TYPE]: transactionType } },
  { range: rangeFilter(start, end) },
  ...uiFiltersES
];

if (transactionName) {
  filters.push({ term: { [TRANSACTION_NAME]: transactionName } });
}
```

Regarding `must` vs `filter`: according to [the docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-bool-query.html) it looks like `must` contributes to the score and is not cachable whereas `filter` is cachable. Not 100% sure but can't see any downsides to using `filter`.

Reply:
agreed that it is more readable in this case! I default to the ternary because TypeScript seems better equipped to deal with it rather than mutating objects. In this case however I don't think it matters.

Reply:
re: filter, I'll create a separate ticket for optimizing all queries.

Reply:
https://github.com/elastic/kibana/issues/41322